openml/flows/flow.py

```diff
@@ -1,80 +1,358 @@
-                     model.__class__.__name__)
-        self.external_version = 'sklearn_' + sklearn.__version__
 
-    def _generate_flow_xml(self):
+        if components is None:
+            components = OrderedDict()
+        elif not isinstance(components, OrderedDict):
+            raise TypeError('components must be of type OrderedDict, '
+                            'but is %s.' % type(components))
+        self.components = components
+
+        if parameters is None:
+            parameters = OrderedDict()
+        elif not isinstance(parameters, OrderedDict):
+            raise TypeError('parameters must be of type OrderedDict, '
+                            'but is %s.' % type(parameters))
+
+        if parameters_meta_info is None:
+            parameters_meta_info = OrderedDict()
+        elif not isinstance(parameters_meta_info, OrderedDict):
+            raise TypeError('parameters_meta_info must be of type OrderedDict, '
+                            'but is %s.' % type(parameters_meta_info))
+
+        keys_parameters = set(parameters.keys())
+        keys_parameters_meta_info = set(parameters_meta_info.keys())
+        if len(keys_parameters.difference(keys_parameters_meta_info)) > 0:
+            raise ValueError('Parameter %s only in parameters, but not in'
+                             'parameters_meta_info.' %
+                             str(keys_parameters.difference(
+                                 keys_parameters_meta_info)))
+        if len(keys_parameters_meta_info.difference(keys_parameters)) > 0:
+            raise ValueError('Parameter %s only in parameters_meta_info, '
+                             'but not in parameters.' %
+                             str(keys_parameters_meta_info.difference(
+                                 keys_parameters)))
+
+        self.parameters = parameters
+        self.parameters_meta_info = parameters_meta_info
+
+        self.external_version = external_version
+        self.uploader = uploader
+
+        if tags is None:
+            tags = []
+        self.tags = tags
+        self.binary_url = binary_url
+        self.binary_format = binary_format
+        self.binary_md5 = binary_md5
+        self.version = version
+        self.upload_date = upload_date
+        self.language = language
+        self.dependencies = dependencies
+        self.flow_id = flow_id
+
+    def _to_xml(self):
         """Generate xml representation of self for upload to server.
 
         Returns
         -------
-        flow_xml : string
+        str
             Flow represented as XML string.
         """
-        model = self.model
+        flow_dict = self.__to_dict()
+        flow_xml = xmltodict.unparse(flow_dict, pretty=True)
+
+        # A flow may not be uploaded with the encoding specification..
+        flow_xml = flow_xml.split('\n', 1)[-1]
+        return flow_xml
+
+    def __to_dict(self):
+        """ Helper function used by _to_xml and __to_dict.
```

Comment:
Yes, since a flow can contain subflows (called components) as for example adaboost it's baseclassifier, these flows need to be encoded as a dictionary as well. Is this a problem?